gremlin-server/src/test/java/org/apache/tinkerpop/gremlin/server/GremlinServerIntegrateTest.java

```diff
@@ -562,4 +568,98 @@ public void shouldBlockWhenMaxConnectionsExceeded() throws Exception {
             cluster.close();
         }
     }
+    
+    @Test
+    public void shouldExecuteInSessionAndSessionlessWithoutOpeningTransactionWithSingleClient() throws Exception {
+        assumeNeo4jIsPresent();
+        
+        final SimpleClient client = new WebSocketClient();
+        
+        //open a transaction, create a vertex, commit
+        final CountDownLatch latch = new CountDownLatch(1);
+        final RequestMessage OpenRequest = RequestMessage.build(Tokens.OPS_EVAL)
+                .processor("session")
+                .addArg(Tokens.ARGS_SESSION, name.getMethodName())
+                .addArg(Tokens.ARGS_GREMLIN, "graph.tx().open()")
+                .create();
+        client.submit(OpenRequest, (r) -> {
+            latch.countDown();
+        });
+        assertTrue(latch.await(1500, TimeUnit.MILLISECONDS));
+        
+        final CountDownLatch latch2 = new CountDownLatch(1);
+        final RequestMessage AddRequest = RequestMessage.build(Tokens.OPS_EVAL)
+                .processor("session")
+                .addArg(Tokens.ARGS_SESSION, name.getMethodName())
+                .addArg(Tokens.ARGS_GREMLIN, "v=graph.addVertex(\"name\",\"stephen\")")
+                .create();
+        client.submit(AddRequest, (r) -> {
+            latch2.countDown();
+        });
+        assertTrue(latch2.await(1500, TimeUnit.MILLISECONDS));
+        
+        final CountDownLatch latch3 = new CountDownLatch(1);
+        final RequestMessage CommitRequest = RequestMessage.build(Tokens.OPS_EVAL)
+                .processor("session")
+                .addArg(Tokens.ARGS_SESSION, name.getMethodName())
+                .addArg(Tokens.ARGS_GREMLIN, "graph.tx().commit()")
+                .create();
+        client.submit(CommitRequest, (r) -> {
+            latch3.countDown();
+            
+        });
+        latch3.await(1500, TimeUnit.MILLISECONDS);
```

Comment:
I can add that.  `SimpleClient` stinks - we need this done: https://issues.apache.org/jira/browse/TINKERPOP3-916